src/lib/components/Modal/Modal.jsx

```diff
@@ -97,9 +97,14 @@ export const Modal = ({
   };
 
   useEffect(() => {
+    const bodyElement = document.getElementsByTagName('body')[0];
+    bodyElement.style = 'overflow-y: hidden';
```

Comment:
I am not happy with this implementation. To be honest, I do not now how to do it properly as this is recommended approach on CSS tricks, but I am afraid this would not work in all cases. 

For example. Some applications scroll whole body and some applications have div with size (top,bottom,left,right=0 or width/height=100%), so this solution would cover only some cases.

@adamkudrna, can you review it and help with this? You are the only one with such knowledge to do it properly, it is too much for me.

Reply:
The problem is that our implementation (where RUI is used) does not have scrollable body, so this change wouldn't help us. We might need to come with another solution :/

Reply:
@bedrich-schindler @adamkudrna The only other viable solution I can see is listening to all scroll events when `Modal` is open and unless the event target is any of the `Modal` children ignore it.

Reply:
Agreed upon on a video call:

* If possible we would like to make all scroll events affect the modal body and not the page underneath regardless of the event source (keyboard, mouse, touchpad etc.). Technical solution is unknown.
* keyboard scroll events must take into account form fields (e.g. `<textarea>`)
* we want to use the `<dialog>` component
* when no active form element or primary button is found and `autofocus` prop is `true` we want to focus the `<dialog>` element

Relevant links:
* https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog#accessibility_considerations
* https://github.com/whatwg/html/wiki/dialog--initial-focus,-a-proposal

Alternate solution is to pass in the page scroll prevention function as a prop.

Reply:
I checked our idols, MUI and Bootstrap 🎉 , and they both prevent body from scrolling. That makes me think:

1. Yes, I'd like to prevent body from scrolling by default.
2. It should opt-out so I can disable it when I don't like it, and…
3. … it could be customizable so I can pass a custom function as @mbohal suggests.

Reply:
Migration to native `<dialog>` element will be implemented in https://github.com/react-ui-org/react-ui/issues/461